spec/jobs/scrape_keyword_job_spec.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe ScrapeKeywordJob do
+  include ActiveJob::TestHelper
+
+  describe '#perform' do
+    context 'given an ERROR occurs' do
+      context "when the keyword doesn't exist" do
+        it 'DOES NOT retry' do
+          perform_enqueued_jobs do
+            described_class.perform_later(keyword_id: 42)
+          end
+
+          assert_performed_jobs 1
+        end
+      end
+
+      context 'when an unexpected error occurs' do
+        it 'retries at most 3 times' do
+          keyword = Fabricate :keyword
+
+          stub = Google::ScrapeService.new(keyword_id: keyword.id)
+          allow(stub).to receive(:call!).and_raise(
+            GoogleScraperRuby::Errors::ScrapeError.new(keyword_id: keyword.id, kind: :unexpected_error)
+          )
+          allow(Google::ScrapeService).to receive(:new).and_return(stub)
```

Comment:
This is a workaround for `allow_any_instance_of` since that one is not recommended by the community.

However, this is just a hack and basically still do the same thing. This test makes sure the background job worker handles errors correctly.

What is your opinion on this?

Reply:
This is normal @Goose97, I don't see any issue with that :)

Reply:
cool. I just realize that, while the overall behavior is the same, the `allow_any_instance_of ` is more intrusive. It also stubs existing instances, whereas the above code only stubs newly created instances. TIL